src/commands.rs

```diff
@@ -2042,93 +2044,118 @@ fn cmd_log(ui: &mut Ui, command: &CommandHelper, args: &LogArgs) -> Result<(), C
+                }
+                if !buffer.ends_with(b"\n") {
+                    buffer.push(b'\n');
+                }
+                if let Some(diff_format) = diff_format {
+                    let mut formatter = ui.new_formatter(&mut buffer);
+                    show_patch(
+                        formatter.as_mut(),
+                        &workspace_command,
+                        &commit,
+                        matcher.as_ref(),
+                        diff_format,
+                    )?;
+                }
+                let node_symbol = if is_checkout { b"@" } else { b"o" };
+                graph.add_node(
+                    &index_entry.position(),
+                    &graphlog_edges,
+                    node_symbol,
+                    &buffer,
                 )?;
             }
-            let node_symbol = if is_checkout { b"@" } else { b"o" };
-            graph.add_node(
-                &index_entry.position(),
-                &graphlog_edges,
-                node_symbol,
-                &buffer,
-            )?;
-        }
-    } else {
-        let iter: Box<dyn Iterator<Item = IndexEntry>> = if args.reversed {
-            Box::new(revset.iter().reversed())
         } else {
-            Box::new(revset.iter())
-        };
-        for index_entry in iter {
-            let commit = store.get_commit(&index_entry.commit_id())?;
-            template.format(&commit, formatter)?;
-            if let Some(diff_format) = diff_format {
-                show_patch(
-                    formatter,
-                    &workspace_command,
-                    &commit,
-                    matcher.as_ref(),
-                    diff_format,
-                )?;
+            let iter: Box<dyn Iterator<Item = IndexEntry>> = if args.reversed {
+                Box::new(revset.iter().reversed())
+            } else {
+                Box::new(revset.iter())
+            };
+            for index_entry in iter {
+                let commit = store.get_commit(&index_entry.commit_id())?;
+                template.format(&commit, formatter)?;
+                if let Some(diff_format) = diff_format {
+                    show_patch(
+                        formatter,
+                        &workspace_command,
+                        &commit,
+                        matcher.as_ref(),
+                        diff_format,
+                    )?;
+                }
+            }
+        }
+    }
+
+    // Check to see if the user might have specified a path when they intended
+    // to specify a revset.
+    if let [only_path] = args.paths.as_slice() {
+        if only_path == "." {
+            // For users of e.g. Mercurial, where `.` indicates the current commit.
+            ui.write_warn(&format!(
+                "warning: The argument {only_path:?} is being interpreted as a path, but this is \
+                 often not useful because all non-empty commits touch '.'.  If you meant to show \
```

Comment:
This is not completely true: if you're in a subdirectory, `.` will match that directory, so it may be useful, and that's actually a somewhat common usecase. You can probably check something like `workspace_command.parse_file_path(only_path)?.is_root()` instead.

Reply:
Or *in addition*, I suppose, since the suggestion won't make sense if you did `cd foo; jj log ..`